core/src/main/java/com/radixdlt/messaging/core/MessageDispatcher.java

```diff
@@ -137,26 +136,77 @@ CompletableFuture<Result<Unit, Cause>> send(final OutboundMessageEvent outboundM
         .findOrCreateChannel(outboundMessage.receiver())
         .thenApply(channel -> send(channel, bytes))
         .thenApply(this::updateStatistics)
-        .exceptionally(t -> completionException(t, receiver, message));
+        .thenApply(
+            result -> {
+              result.onError(err -> logSendError(message, receiver, err.toString()));
+              return result;
+            })
+        .exceptionally(
+            t -> {
+              final var cause = t.getCause() != null ? t.getCause().getMessage() : t.getMessage();
+              logSendError(message, receiver, cause);
+              return IO_ERROR.result();
+            });
+  }
+
+  private void logSendError(Message message, NodeId receiver, String cause) {
+    synchronized (sendErrorLogRateLimitersByReceiver) {
+      final RejectionCountingRateLimiter rateLimiter;
+      if (sendErrorLogRateLimitersByReceiver.contains(receiver)) {
+        rateLimiter = sendErrorLogRateLimitersByReceiver.get(receiver).orElseThrow();
+      } else {
+        rateLimiter = new RejectionCountingRateLimiter(0.05);
```

Comment:
What does this 0.05 mean? We can have 20 logs per second per validator? Seems excessive still?

Or maybe it's one every 20 seconds? Perhaps we make it  once a minute (eg 1.0 / 60.0) and add a comment that explains it in human terms? OR make the RateLimiter take a TimeSpan instead? (or whatever the Java equivalent is?)

Reply:
:+1: will change to once a minute